src/PresentationalComponents/Inventory/Hooks/useBulkSelect/useBulkSelect.js

```diff
@@ -116,7 +108,7 @@ const useBulkSelect = ({
                 : []),
             ],
             checked,
-            onSelect: !isDisabled ? selectPage : undefined,
+            onSelect: selectedIds?.length > 0 ? () => clear() : selectAll,
```

Comment:
How about just passing:
onSelect: selectedIds?.length > 0 ? () => clear() : selectAll,
```suggestion
            onSelect: selectedIds?.length > 0 ? clear : selectAll,
```